### PR TITLE
added father, grand_father, and birth_place in scan document button

### PR DESCRIPTION
**Pull request by reichie020212**

### CODE REVIEW

File: spp_scan_id_document/tests/test_registrant.py [hunk 1]
```diff
@@ -38,4 +38,3 @@ def test_on_scan_id_document_details_success(self):
 
         self.assertEqual(self.applicant.family_name, "Red")
         self.assertEqual(self.applicant.given_name, "Blue")
-        self.assertEqual(self.applicant.gender, "Male")
```
Comment (nhatnm0612) [inline spp_scan_id_document/tests/test_registrant.py:41]: @reichie020212 why did you remove this line? I meant: the data is there, it should work 😕

Comment (reichie020212) [reply spp_scan_id_document/tests/test_registrant.py:41]: @nhatnm0612 to be honest, I don't know why this test is not working on the job here in the link (https://github.com/OpenSPP/openspp-registry/actions/runs/4980432358/jobs/8913336844?pr=235) but when I run it in my local it is working. That's the reason why I decided to just remove this line of code.

**Review by emjay0921 — DISMISSED** [review]

File: spp_scan_id_document/tests/test_registrant.py (hunk 1 above)
Comment (nhatnm0612) [reply spp_scan_id_document/tests/test_registrant.py:41]: @reichie020212 `father_name` is defined in another module [which depends on this module]. On this test run, test cases start right after the module installed and at that time, `res_partner` table does not have this column yet, also Individual instance does not recognize this field yet to write. I think you should remove this logic out of PR, coz `pds-spp` is not our product, this one is.

Comment (reichie020212) [reply spp_scan_id_document/tests/test_registrant.py:41]: @nhatnm0612 noted on that

File: spp_scan_id_document/model/registrant.py [hunk 2]
```diff
@@ -27,10 +27,13 @@ def on_scan_id_document_details(self):
                     vals = {
```
Comment (nhatnm0612) [inline spp_scan_id_document/model/registrant.py]: @reichie020212 I think we should take this part out to be a hook method, so we can inherit that easier in the later modules

```python
def _prepare_update_vals(self, details):
    return {...}
```

then this line will become
```python
vals = self._prepare_update_vals(details)
```

so, in later module, it will be:
```python
def _prepare_update_vals(self, details):
    res = super(). _prepare_update_vals(details)
    res.update(...)
    return res
```

File: spp_scan_id_document/tests/test_registrant.py (hunk 1 above)
Comment (reichie020212) [reply spp_scan_id_document/tests/test_registrant.py:41]: @nhatnm0612 done updating the code. kindly check again. Thanks.

**Review by nhatnm0612 — DISMISSED** [review]

**Review by nhatnm0612 — APPROVED** [review]

**Review by gonzalesedwin1123 — APPROVED** [review]
